src/components/FileHistoryElement.tsx

```diff
@@ -14,85 +19,125 @@ export function FileHistoryElement(props: props) {
 }
 
 export function CommitDistFragment(props: CommitDistFragProps) {
-  if (!props.show) return null
+  if (!props.items) return null
+  const sortMethod: SortCommitsMethods = props.sortBy !== undefined ? props.sortBy : "date"
+
+  const cleanGroupItems: { [ key: string ]: GitLogEntry[] } = sortCommits(props.items, sortMethod)
+
+  const items: Array<AccordionData> = new Array<AccordionData>()
+  for (const [ key, values ] of Object.entries(cleanGroupItems)) {
+    items.push({
+      title: key,
+      content: (
+        <>
+          { values.map((value: GitLogEntry) => {
+            return (
+              <>
+                <li
+                  className="cursor-auto"
+                  style={{listStyleImage: `url(${commitIcon})`}}
+                  onClick={ () => (props.handleOnClick ? props.handleOnClick(value) : null) }
+                  key={ value.time + value.message + "--itemContentAccordion" }
+                >
+                  { value.message }
+                </li>
+              </>
+            )
+          }) }
+        </>
+      ),
+    })
+  }
 
   return (
-    <>
-      {props.items.map((commit) => {
-        return (
-          <Fragment key={commit.time.toString() + commit.message}>
-            <span
-              className="overflow-hidden overflow-ellipsis whitespace-pre text-sm font-bold opacity-80"
-              title={commit.message + " (" + commit.author + ")"}
-            >
-              {commit.message}
-            </span>
-            <p className="break-all text-right text-sm">{dateFormatLong(commit.time)}</p>
-          </Fragment>
-        )
-      })}
-    </>
+      <Fragment key={ items.length.toString() + sortMethod + props.commitCutoff.toString() + new Date().toDateString() }>
+        <Accordion
+          titleLabels={ true }
+          multipleOpen={ true }
+          openByDefault={ true }
+          items={ items }
+          itemsCutoff={ props.commitCutoff }
+        ></Accordion>
+      </Fragment>
   )
 }
 
 function CommitHistory(props: { commits: GitLogEntry[] | undefined }) {
   const commitHistoryExpandId = useId()
   const [collapsed, setCollapsed] = useState<boolean>(true)
   const commits = props.commits ?? []
-  const commitCutoff = 2
+  const commitCutoff = 3
 
-  const lessThanCutOff = commits.length <= commitCutoff + 1
-  if (lessThanCutOff) {
+  if (commits.length == 0) {
     return (
       <>
         <h3 className="font-bold">Commit history</h3>
-        <div className="grid grid-cols-[1fr,auto] gap-x-1 gap-y-1.5">
-          {commits.length > 0 ? <CommitDistFragment show={true} items={commits} /> : <p>No commits found</p>}
+        <div>
+          <p>No commits found</p>
         </div>
       </>
     )
   }
   return (
     <>
       <div className="flex cursor-pointer justify-between hover:opacity-70">
-        <label className="label grow" htmlFor={commitHistoryExpandId}>
+        <label className="label grow">
```

Comment:
Removing this makes it such that clicking the title does not expand / contract the extra commit dates, so I would suggest to revert this change.